Build failure: pkgsMusl.gdb

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

nix build nixpkgs/master#pkgsMusl.gdb

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
CXX    objc-lang.o
  CXX    objfiles.o
  CXX    obsd-tdep.o
  CXX    observable.o
In file included from ../../gdb/nds32-tdep.c:43:
../../gdb/../include/opcode/nds32.h:26:9: error: ‘REG_R8’ redefined [-Werror]
   26 | #define REG_R8          (8)
      |         ^~~~~~
In file included from /nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/signal.h:48,
                 from ../gnulib/import/signal.h:52,
                 from ../gnulib/import/sys/select.h:118,
                 from /nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/sys/time.h:9,
                 from ../gnulib/import/sys/time.h:39,
                 from ../gnulib/import/sys/select.h:89,
                 from /nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/sys/types.h:71,
                 from ../gnulib/import/sys/types.h:39,
                 from ../gnulib/import/stdio.h:58,
                 from ../../gdb/../gdbsupport/common-defs.h:103,
                 from ./../../gdb/defs.h:26,
                 from <command-line>:
/nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/bits/signal.h:11:9: note: this is the location of the previous definition
   11 | #define REG_R8 REG_R8
      |         ^~~~~~
../../gdb/../include/opcode/nds32.h:27:9: error: ‘REG_R10’ redefined [-Werror]
   27 | #define REG_R10         (10)
      |         ^~~~~~~
/nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/bits/signal.h:15:9: note: this is the location of the previous definition
   15 | #define REG_R10 REG_R10
      |         ^~~~~~~
../../gdb/../include/opcode/nds32.h:28:9: error: ‘REG_R12’ redefined [-Werror]
   28 | #define REG_R12         (12)
      |         ^~~~~~~
/nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/bits/signal.h:19:9: note: this is the location of the previous definition
   19 | #define REG_R12 REG_R12
      |         ^~~~~~~
../../gdb/../include/opcode/nds32.h:29:9: error: ‘REG_R15’ redefined [-Werror]
   29 | #define REG_R15         (15)
      |         ^~~~~~~
/nix/store/hsvrmvp1i7k326fpvdrg99gmka863fwi-musl-1.2.5-dev/include/bits/signal.h:25:9: note: this is the location of the previous definition
   25 | #define REG_R15 REG_R15
      |         ^~~~~~~
  CXX    opencl-lang.o
  CXX    or1k-linux-tdep.o
  CXX    or1k-tdep.o
  CXX    osabi.o
  CXX    osdata.o
```

### Additional context

_No response_

### System metadata


 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.4, NixOS, 26.05 (Yarara), 26.05.20260108.41322ef`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.95.0-pre20260108-dev_851c4e3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/yuka/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/yuka/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/yuka/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/yuka/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/ayki914v8yd5iprzg54rz7z010rlwvc2-lix-2.95.0-pre20260108-dev_851c4e3/share`
 - nixpkgs: `/nix/store/mycwi1428s323w5v6akwrnqk21yqy9z0-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gd/gdb/package.nix
+++ b/pkgs/by-name/gd/gdb/package.nix
@@ -5,6 +5,7 @@
 
   # Build time
   fetchurl,
+  fetchpatch,
   pkg-config,
   perl,
   texinfo,
@@ -86,6 +87,17 @@ stdenv.mkDerivation (finalAttrs: {
 
   patches = [
     ./debug-info-from-env.patch
+
+    (fetchurl {
+      name = "musl.patch";
+      url = "https://inbox.sourceware.org/gdb-patches/20260324164527.1446549-2-sunilkumar.dora@windriver.com/raw";
+      hash = "sha256-FC4DDVS4wtE/HXtbUqvkxu9+e7nE3DYi1zIuQP9yQO8=";
+    })
+    (fetchpatch {
+      name = "musl-aarch64.patch";
+      url = "https://sourceware.org/git/?p=binutils-gdb.git;a=patch;h=1ccc3f6a2e28fa1f3357826374cba165b3ba3ff7";
+      hash = "sha256-Q2oTo2b+9yNN3PSsxqgxV4/9/05uFE/JMLe1CPs9Y7I=";
+    })
   ]
   ++ optionals stdenv.hostPlatform.isDarwin [
     ./darwin-target-match.patch

